### ISSUE TITLE
Empty lists do not evaluate to `true` with `empty?`

### ISSUE THREAD

**Issue by bugabinga** [nushell/nushell#3903]
### Describe the bug

According to [the book](https://www.nushell.sh/book/working_with_lists.html) an empty table piped to `empty?` should return `true`.
However it seems to print an empty line instead.

### How to reproduce

```sh
let colors = []
$colors | empty?
```

or simply

```sh
[] | empty?
```

### Expected behavior

`[] | empty?` should print `true`.

### Configuration

version | pivot | to md
|Column0|Column1|
|-|-|
|version|0.35.0|
|build_os|windows-x86_64|
|rust_version|rustc 1.54.0 (a178d0322 2021-07-26)|
|rust_channel|stable-x86_64-pc-windows-msvc (default)|
|cargo_version|cargo 1.54.0 (5ae8d74b3 2021-06-22)|
|pkg_version|0.35.0|
|build_time|2021-08-04 10:11:37|
|build_rust_channel|release|
|features|clipboard-cli, ctrlc, dataframe, default, rustyline, term, trash, uuid, which, zip|
|installed_plugins|binaryview, chart bar, chart line, fetch, from bson, from sqlite, inc, match, post, ps, query json, s3, selector, start, sys, textview, to bson, to sqlite, tree, xpath|

### Additional context

Other datatypes work as expected with `empty?`.

```sh
∽
  1 | empty?
false

∽
  "" | empty?
true

∽
  $true | empty?
false

∽
  $false | empty?
false

∽
  0 | empty?
false

∽
  () | empty?
true

∽
  $nothing | empty?
true

∽
  0.0 | empty?
false

∽
  0sec | empty?
false

∽
  0..0 | empty?
false

∽
  0b | empty?
false

∽
  [[];[]] | empty?
true

∽
  {} | empty?
true
```

Notice the table type in particular .
An empty table works: `[[];[]]`.
But an empty list seemingly not: `[]`.

**Comment by jafriyie1** [comment]
@fdncred Is anyone working on this? I'd like to give this a shot if no one is.

**Comment by fdncred** [comment]
Go for it @jafriyie1

**Comment by fdncred** [comment]
Similar issue #3929

**Comment by fdncred** [comment]
@jafriyie1 how's it going on this issue? We've had requests similar to yours here recently.

**Comment by jafriyie1** [comment]
@fdncred I stepped away a while from this as I got busy with other things. my apologies. I made a test for the corrected behavior and I am noticing that the issue comes from here [empty_code](https://github.com/nushell/nushell/blob/ead4029d494c770cae97a11c470aee326a5ba33b/crates/nu-command/src/commands/filters/empty.rs#L83). When I was testing around with this I saw that `[] | empty?` returns "" instead of true or false. 

In addition to this, when we pass in `([] | empty?) | empty?` we get the expected boolean. This makes me think that the parser for some reason treats an empty list `[]` differently than other values. Because of this, I'm wondering how I should go about fixing this. Should we just make an explicit match on `[]`? Or is there a more elegant approach to this?
